File: Data Structures & Algorithms/longest-consecutive-sequence/submission-8.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        
        seen = set(nums)
        longest = 0

        for n in seen:
            if (n - 1) not in seen: 
                current = 1
                
                while n + current in seen:
                    current += 1
                
                longest = max(current, longest)

        return longest
        

File: Data Structures & Algorithms/two-integer-sum/submission-8.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const seen = {};

        for (let i = 0; i < nums.length; i++){
            const comp = target - nums[i];

            if(seen[comp] >= 0) {
                return [seen[comp], i] 
            }

            seen[nums[i]] = i;
        } 
        
        return [];
    }
}
